src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,137 @@
 package core.basesyntax;
 
+import java.util.Map;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size = 0;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
 
+        if (size >= table.length * DEFAULT_LOAD_FACTOR) {
+            resize(table);
+        }
+
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (node.hash == hash && Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        addNode(hash, key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (node.hash == hash && Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static final int hash(Object key) {
+        int h;
+        return (key == null) ? 0 : (h = key.hashCode()) ^ (h >>> 16);
+    }
+
+    private void addNode(int hash, K key, V value, int index) {
+        Node<K, V> newNode = new Node<>(hash, key, value, table[index]);
```

Comment:
When using the `Node` constructor, ensure the `next` parameter is set correctly to maintain the linked list structure in case of collisions. This is crucial for the correct functionality of the hash map.